src/commands/MakeRoomAdminCommand.ts

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2021 Marco Cirillo
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Mjolnir } from "../Mjolnir";
+import { RichReply } from "matrix-bot-sdk";
+
+// !mjolnir make admin <room> [<user ID>]
+export async function execMakeRoomAdminCommand(roomId: string, event: any, mjolnir: Mjolnir, parts: string[]) {
+    const isAdmin = await mjolnir.isSynapseAdmin();
+    if (!isAdmin) {
```

Comment:
Could you also add a setting to enable/disable this feature?
You'll need to patch `config.ts`, `default.yaml` and `harness.yaml`.

Let's create a setting

```ts
admin?: {
   makeRoomAdminCommand: boolean;
}
```

I'm sure that we'll want to use this `admin` object for other reasons.

Reply:
I can do that, but considering it's a command and not some protection that is pluggable am I supposed to hide it from the command list and do conditional loading of the import in the command handler if it's not activated in the config?

Reply:
That would be the ideal case, but I feel that this is beyond the scope of your PR. One of these days, we'll refactor commands to make them easier to add/remove.

For the time being, what about rejecting from within `execMakeRoomAdminCommand`, just as you do when `!isAdmin`?

Something like

```ts
if (!config.?admin?.enableMakeRoomAdminCommand) {
   // ...reject
}
```